File: Ahorn/entities/PopstarBerry.jl
module PopstarBerry

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/PopstarBerry" PopstarBerry(x::Integer, y::Integer, collectSound::String="Elaborate", customCollectSound::String="", levelSet::String="Maggy/DESOLO_ZANTAS/19-Space", maps::String="", requires::String="")

const placements = Ahorn.PlacementDict(
    "PopstarBerry" => Ahorn.EntityPlacement(PopstarBerry)
)

function Ahorn.selection(entity::PopstarBerry)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::PopstarBerry, room::Maple.Room)
    Ahorn.drawSprite(ctx, "collectables/maggy/popstarberry/Maggy/DesoloZantas/spin/000", entity.x, entity.y)
end

end

File: Loenn/entities/popstarBerry.lua
local popstarBerry = {}

popstarBerry.name = "MaggyHelper/PopstarBerry"
popstarBerry.depth = -100
popstarBerry.texture = "collectables/popstarberry/spin/000"

popstarBerry.fieldInformation = {
    collectSound = {
        editable = false,
        options = {
            ["Original"]   = "Original",   -- strawberry_get + strawberry_blue_touch
            ["Elaborate"]  = "Elaborate",   -- strawberry_get + Maggy_DesoloZantas/game/general/strawberry_get
            ["Minimalist"] = "Minimalist",  -- strawberry_get only
            ["Custom"]     = "Custom"       -- strawberry_get + customCollectSound event path
        }
    },
    customCollectSound = {
        editable = true  -- visible/active only when collectSound == "Custom"
    }
}

popstarBerry.placements = {
    name = "PopstarBerry",
    data = {
        collectSound       = "Elaborate",
        customCollectSound = "",
        levelSet           = "Maggy/DESOLO_ZANTAS/19-Space",
        maps               = "",
        requires           = ""
    }
}

return popstarBerry
